### PR TITLE
clang-tidy: fix mismatched variable names

### PR DESCRIPTION
**Pull request by neheb**
Found with readability-inconsistent-declaration-parameter-name

### CODE REVIEW

**Comment by sbooth** [comment]
In principle I agree that the variable names for functions should have the same name in the header as in the implementation. The issue with this PR is that by changing the variable names in the header to match those in the implementation, the doxygen comments are no longer correct.

**Comment by neheb** [comment]
No idea what you mean. -Werror=documentation does not throw errors.

File: taglib/asf/asftag.h [hunk 1]
```diff
@@ -101,48 +101,48 @@ namespace TagLib {
       /*!
        * Sets the title to \a s.
        */
-      void setTitle(const String &s) override;
+      void setTitle(const String &value) override;
```
Comment (sbooth) [inline taglib/asf/asftag.h:105]: The documentation comment refers to `s` which is the old variable name.

**Comment by sbooth** [comment]
> No idea what you mean. -Werror=documentation does not throw errors.

I added an illustrative comment.

File: taglib/asf/asftag.h (hunk 1 above)
Comment (neheb) [reply taglib/asf/asftag.h:105]: that doesn't look like a proper Doxygen comment.

Comment (sbooth) [reply taglib/asf/asftag.h:105]: I'm not sure what constitutes proper vs. improper doxygen comments but the variable names need to match.

Comment (neheb) [reply taglib/asf/asftag.h:105]: ah nvm. Seems it is. I'll fix them. Oddly enough clang warns only on \param and not \a.

**Comment by neheb** [comment]
Fixed.

File: taglib/asf/asftag.h [hunk 2]
```diff
@@ -184,8 +184,8 @@ namespace TagLib {
       AttributeList attribute(const String &name) const;
 
       /*!
-       * Sets the \a key attribute to the value of \a attribute. If an attribute
-       * with the \a key is already present, it will be replaced.
+       * Sets the \a attribute attribute to the value of \a attribute. If an attribute
```
Comment (sbooth) [inline taglib/asf/asftag.h:187]: `attribute` here should be either `name` or `key`

Comment (neheb) [reply taglib/asf/asftag.h]: I think this is what you meant.

Comment (sbooth) [reply taglib/asf/asftag.h]: I think it should be `Sets the \a name attribute to the value of \a attribute` since `name` is the dictionary key.

Comment (neheb) [reply taglib/asf/asftag.h]: Fixed

File: taglib/mpeg/id3v2/frames/commentsframe.h [hunk 3]
```diff
@@ -104,7 +104,7 @@ namespace TagLib {
        *
        * \see language()
        */
-      void setLanguage(const ByteVector &languageCode);
+      void setLanguage(const ByteVector &languageEncoding);
```
[the remark this answers is not in the document]
Comment (neheb) [reply taglib/mpeg/id3v2/frames/commentsframe.h:107]: Fixed.

File: taglib/ape/apeitem.h [hunk 4]
```diff
@@ -191,7 +191,7 @@ namespace TagLib {
        *
        * \see ItemTypes
        */
-      void setType(ItemTypes type);
+      void setType(ItemTypes val);
```
[the remark this answers is not in the document]
Comment (neheb) [reply taglib/ape/apeitem.h:194]: Fixed.

File: taglib/mpeg/id3v2/frames/synchronizedlyricsframe.h [hunk 5]
```diff
@@ -182,7 +182,7 @@ namespace TagLib {
        *
        * \see language()
        */
-      void setLanguage(const ByteVector &languageCode);
+      void setLanguage(const ByteVector &languageEncoding);
```
[the remark this answers is not in the document]
Comment (neheb) [reply taglib/mpeg/id3v2/frames/synchronizedlyricsframe.h:185]: Fixed.

File: taglib/mpeg/id3v2/frames/unsynchronizedlyricsframe.h [hunk 6]
```diff
@@ -102,7 +102,7 @@ namespace TagLib {
        *
        * \see language()
        */
-      void setLanguage(const ByteVector &languageCode);
+      void setLanguage(const ByteVector &languageEncoding);
```
[the remark this answers is not in the document]
Comment (neheb) [reply taglib/mpeg/id3v2/frames/unsynchronizedlyricsframe.h:105]: Fixed.

**Pull request merged**